### PR TITLE
Week 1 Notes

### PR DESCRIPTION
**Pull request by demanr**
Here are my notes for week 1 in Markdown, with added links.

### CODE REVIEW

**Comment by JacquesCarette** [comment]
This is fantastic. I have a number of corrections to make, so I'll do a full review soon, and then merge in this PR.

I would happily accept more PRs like this (for bonus grades, naturally - it is extra work).

File: Lecture/Week1Lecture1&2Notes.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	- used three way "if" statement (had 3 places to jump)
+		- if a variable is less than 0 one option, if the variable = 0 a 2nd choice, if the variable is greater than 0 a 3rd option is chosen
+		- it was done this way as the hardware had 3 way jump instruction
+	- all about scientific computation
+- [Wikipedia Page](https://en.wikipedia.org/wiki/Fortran)
+### lisp
+- soon after fortran came lisp
+	- more for computing abstract or symbolic stuff
+	- used for "oldies AI" (very basic)
+	- dynamic scope for if statements (fortran had static)
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: dynamic scope for variables, not if statements. See https://prl.khoury.northeastern.edu/blog/2019/09/05/lexical-and-dynamic-scope/ and https://en.wikipedia.org/wiki/Scope_(computer_science)

File: Lecture/Week1Lecture1&2Notes.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	- lots of these programs needed to b fixed
+	- COBOL programmers made a lot of money in this times
+- made to read like english 
+	- "add one to x" is a valid COBOL command
+- [Wikipedia Page](https://en.wikipedia.org/wiki/COBOL)
+### Algol 60
+- influenced many languages, designed by a committee
+- first language with a standard
+- [Wikipedia Page](https://en.wikipedia.org/wiki/ALGOL_60)
+- [Original Website](https://algol60.org/)
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: should also link to the Algol page

File: Lecture/Week1Lecture1&2Notes.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+### PL/I ("PL one")
+- many features, but they weren't very thought through
+- VERY BAD compile efficiency
+- described as an "awful langage",  can accomplish things but very mishmashed
+- [Wikipedia Page](https://en.wikipedia.org/wiki/PL/I)
+- [PL/I IBM documentation](https://www.ibm.com/docs/en/zos-basic-skills?topic=zos-pli)
+### Smalltalk
+- Used a lot of OOP
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: Is the first truly OO language, and most ideas of OO that were not already in Simula 67 come from here

File: Lecture/Week1Lecture1&2Notes.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+- [Rust Controversy](https://www.looper.com/1209567/the-entire-rust-shooting-controversy-explained/)
+- [Interesting Reddit discussion on the Rust controversy](https://www.reddit.com/r/rust/comments/12lb0am/can_someone_explain_to_me_whats_happening_with/)
+- [Rust guide book (free)](https://doc.rust-lang.org/stable/book/)
+# D
+- An improved version of C
+- [Wikipedia Page](https://en.wikipedia.org/wiki/D_(programming_language)
+- [D language Website](https://dlang.org/)
+# Forth
+- influenced PDF (portable document format)
+- old but still used today
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: mostly for embedded software

File: Lecture/Week1Lecture1&2Notes.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+- The newer version, SQL, is turing complete
+	- not necessarily good as you can no longer calculate how long it takes to execute a query
+	- It is a reason why not many people use SEQUEL instead
+- [Wikipedia Page](https://en.wikipedia.org/wiki/SQL)
+# SH
+- SH stands for shell
+	- despite common belief it is a full programming language! 
+- Microsoft powershell does NOT look like sh, it was a complete redesign
+- [Wikipedia Page](https://en.wikipedia.org/wiki/Shell_script)
+# OS2
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: OS/2 is an operating system, not a programming language. That part isn't testable material, that was just a story from my past.

File: Lecture/Week1Lecture1&2Notes.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+- Primary data structure is arrays 
+- Strange syntax
+- [Wikipedia Page](https://en.wikipedia.org/wiki/APL_(programming_language)
+- [Try APL yourself here](https://tryapl.org/)
+# SaC
+- descendant of APL, stands for Single assignment C
+- Purely functional
+- [Wikipedia Page](https://en.wikipedia.org/wiki/SAC_programming_language)
+- [SaC Website](https://www.sac-home.org/index)
+# Snowbol
```
Comment (JacquesCarette) [inline Lecture/Week1Lecture1&2Notes.md]: Snobol

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
A few things to fix - thanks again.

File: Lecture/Week1Lecture1&2Notes.md (hunk 1 above)
Comment (demanr) [reply Lecture/Week1Lecture1&2Notes.md]: Good to know, thanks

File: Lecture/Week1Lecture1&2Notes.md (hunk 2 above)
Comment (demanr) [reply Lecture/Week1Lecture1&2Notes.md]: I couldn't find an Algol page, could you please elaborate a bit on what you're looking for?

Comment (JacquesCarette) [reply Lecture/Week1Lecture1&2Notes.md]: https://en.wikipedia.org/wiki/ALGOL

Comment (demanr) [reply Lecture/Week1Lecture1&2Notes.md]: Thanks!

**Comment by demanr** [comment]
Should be good now

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**